cdap-data-fabric/src/main/java/co/cask/cdap/data2/sql/PostgresSqlStructuredTable.java

```diff
@@ -129,6 +129,28 @@ public void upsert(Collection<Field<?>> fields) throws InvalidFieldException, IO
     }
   }
 
+  @Override
+  public CloseableIterator<StructuredRow> scan(Field<?> index) throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: Scan index {}", tableSchema.getTableId(), index);
+    fieldValidator.validateField(index);
+    if (!tableSchema.isIndexColumn(index.getName())) {
+      throw new InvalidFieldException(tableSchema.getTableId(), index.getName(), "is not an indexed column");
+    }
+
+    String sql = getReadQuery(Collections.singleton(index), Collections.emptySet(), false);
```

Comment:
I feel like it would be cleaner to pass in null for the columns to read. 
emptySet() is like saying, I want no columns except for the primary keys.

Reply:
+1